src/UhdmAst.cpp

```diff
@@ -1046,6 +1046,86 @@ AstNode* process_operation(vpiHandle obj_h, UhdmShared& shared) {
     return nullptr;
 }
 
+AstVar* process_parameter(vpiHandle obj_h, UhdmShared& shared, bool get_value) {
+    AstVar* parameterp = nullptr;
+    AstNode* parameterValuep = nullptr;
+
+    std::string objectName;
+    for (auto name : {vpiName, vpiFullName, vpiDefName}) {
+        if (auto s = vpi_get_str(name, obj_h)) {
+            objectName = s;
+            sanitize_str(objectName);
+            break;
+        }
+    }
+
+    if (is_imported(obj_h)) {
+        // Skip imported parameters, they will still be visible in their packages
+        UINFO(3, "Skipping imported parameter " << objectName << std::endl);
+        return nullptr;
+    }
+
+    AstRange* rangeNodep = nullptr;
+    visit_one_to_many({vpiRange}, obj_h, shared, [&](AstNode* nodep) {
+        if (nodep) rangeNodep = reinterpret_cast<AstRange*>(nodep);
+    });
+
+    AstNodeDType* dtypep = nullptr;
+    auto typespec_h = vpi_handle(vpiTypespec, obj_h);
+    if (typespec_h) { dtypep = getDType(typespec_h, shared); }
+
+    // If no typespec provided assume default
+    if (dtypep == nullptr) {
+        dtypep = new AstBasicDType(new FileLine("uhdm"),
```

Comment:
We're making an assumption about the type here, there should be at least a `UINFO` to alert the user.

How often is the type missing with current UHDM/Surelog?

Reply:
I added UINFO in line 1079. 
Now the type is missing  for 19 distinct parameter names.